### PR TITLE
tape recorder v3 revival

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
revival of #30089
refactored, microoptimised and addressing reviews
changed the weird dual-state thing to just be mode and thats it

stores the verb used, but the exact string the server picked in SendEntitySpeak can't be saved. presumably this will be possible in chat refactor, then the tape can store the exact message (including whispering, which still isnt stored right)

## Why / Balance
see old pr

## Technical details
Active bool changed to ActiveTapeRecorder so it can be queried for
UI uses RadioOptions instead of making them manually
most of the state is just gotten from the component instead of being sent in every bui state
it also stores the used speech verb now

## Media
Pause | Record | Playback | Rewind

it will use a random string from the speech verb that was recorded

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
:cl: BombasterDS, deltanedas
- add: Tape recorders and cassettes can now be crafted in autolathes for recording crew speech.

### CODE REVIEW

**Comment by lzk228** [comment]
he is cooking 🗣️🗣️🗣️

**Comment by MemesHodachy** [comment]
let him cook dat 🔥

**Comment by deltanedas** [comment]
the whisper stuff from #17898 needs to be fixed with a chat refactor, right now if i want to have it i would need to copy paste all of its logic to hide identity in the transcript

**Comment by beck-thompson** [comment]
Fourth times the charm! Good luck 🫡

**Comment by slarticodefast** [comment]
Will need to be adjusted for #30798

**Comment by UrPrice** [comment]
@metalgearsloth heyo, is it ready for review or merge? 3rd iteration of tape recorder, we soon start merge for us earlier than it will be merge to upstream

File: Content.Shared/TapeRecorder/SharedTapeRecorderSystem.cs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    {
+        var (uid, comp) = ent;
+        if (!_ui.IsUiOpen(uid, TapeRecorderUIKey.Key))
+            return;
+
+        var hasCassette = TryGetTapeCassette(ent, out var tape);
+        var hasData = false;
+        var currentTime = 0f;
+        var maxTime = 0f;
+        var cassetteName = "Unnamed";
```
Comment (stalengd) [inline Content.Shared/TapeRecorder/SharedTapeRecorderSystem.cs:398]:
```suggestion
        var cassetteName = Loc.GetString("tape-recorder-menu-cassette-unnamed");
```

File: Resources/Locale/en-US/taperecorder/taperecorder.ftl [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+tape-recorder-menu-title = Tape Recorder
+tape-recorder-menu-controls-label = Controls:
+tape-recorder-menu-stopped-button = Pause
+tape-recorder-menu-recording-button = Record
+tape-recorder-menu-playing-button = Playback
+tape-recorder-menu-rewinding-button = Rewind
+tape-recorder-menu-print-button = Print record transcript
+tape-recorder-menu-cassette-label = Cassette tape: {$cassetteName}
+tape-recorder-menu-no-cassette-label = Cassette tape is not inserted
+
```
Comment (stalengd) [inline Resources/Locale/en-US/taperecorder/taperecorder.ftl:24]:
```suggestion
tape-recorder-menu-no-cassette-label = Cassette tape is not inserted
tape-recorder-menu-cassette-unnamed = Unnamed
```

**Review by stalengd — CHANGES_REQUESTED** [review]
Pretty sure this should be localized

**Comment by Strol20** [comment]
Will this be merged after the chat refactoring?

**Comment by Strol20** [comment]
Sorry for my English

**Comment by deltanedas** [comment]
guess maints dont want it, using it downstream then

**Comment by Strol20** [comment]
Please open pr. Sooner or later someone would check and merge.

**Comment by deltanedas** [comment]
i dont want to deal with merge conflicts for another year when nobody cares

**Comment by Vonsant** [comment]
I feel so sad right now.

**Comment by NakataRin** [comment]
With all my heart I wish for this to be implemented. But yet again, no luck

**Comment by deltanedas** [comment]
notice how nobody begging for it is a maintainer... :trollface:

**Comment by JIPDawg** [comment]
> notice how nobody begging for it is a maintainer... :trollface:

What if you became that maintainer.

**Comment by dffdff2423** [comment]
the day deltanedas creates a discord account is the day the world ends.

**Pull request closed without merging**